platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/tipset/TipsetTrackerTests.java

```diff
@@ -141,17 +146,18 @@ void basicBehaviorTest() {
             assertTipsetEquality(addressBook, expectedTipsets.get(fingerprint), tracker.getTipset(fingerprint));
         }
 
-        // Slowly advance the minimum generation, we should see tipsets disappear as we go.
-        long minimumGenerationNonAncient = 0;
+        // Slowly advance the ancient threshold, we should see tipsets disappear as we go.
+        long ancientThreshold = ancientMode == AncientMode.BIRTH_ROUND_THRESHOLD
+                ? ConsensusConstants.ROUND_FIRST
+                : EventConstants.FIRST_GENERATION;
         while (tracker.size() > 0) {
-            minimumGenerationNonAncient += random.nextInt(1, 5);
-            // FUTURE WORK: change test to use birthRound number instead of minimum generation non-ancient.
-            final NonAncientEventWindow nonAncientEventWindow = new NonAncientEventWindow(
-                    1, minimumGenerationNonAncient, 0 /* ignored in this context */, AncientMode.GENERATION_THRESHOLD);
+            ancientThreshold += random.nextInt(1, 5);
+            final NonAncientEventWindow nonAncientEventWindow =
+                    new NonAncientEventWindow(1, ancientThreshold, 1 /* ignored in this context */, ancientMode);
```

Comment:
Here is another case of allegedly ignored, but value is changed

Reply:
the value is ignored for the unit test, but the data structure still has an integrity check when the ancient mode is for birth round.